bookstore/clone.py

```diff
@@ -40,6 +37,68 @@ class BookstoreCloneHandler(IPythonHandler):
     `Jupyter Notebook reference on Custom Handlers <https://jupyter-notebook.readthedocs.io/en/stable/extending/handlers.html#registering-custom-handlers>`_
     """
 
+    def initialize(self):
+        """Helper to retrieve bookstore setting for the session."""
+        self.bookstore_settings = BookstoreSettings(config=self.config)
+
+    @web.authenticated
+    async def get(self):
+        """GET /bookstore/clone?s3_bucket=<your_s3_bucket>&s3_key=<your_s3_key>
+
+        Renders an options page that will allow you to clone a notebook
+        from a specific bucket via the Bookstore cloning API.
+        """
+        s3_bucket = self.get_argument("s3_bucket")
+        if s3_bucket == '' or s3_bucket == "/":
+            raise web.HTTPError(400, "Must have a bucket to clone from")
```

Comment:
```suggestion
            raise web.HTTPError(400, "Requires an S3 bucket")
```

Reply:
I think I'd like to indicate that this is specifically an issue around cloning, because it does not require the s3_bucket to be set from the settings, and so should be distinguished from the issues that arise around working without an s3_bucket on the archiving/publishing side.

Reply:
How about my suggestion with `for clone` or `to clone from`

Reply:
I just pushed changes for "in order to clone" I hadn't seen your response yet because in between writing the change and pushing it i was writing the long response to your question about s3_key

Reply:
Sorry about that!